Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -56,20 +56,46 @@
 #include <wtf/text/AtomStringHash.h>
 #include <wtf/text/StringToIntegerConversion.h>
 
+static void videoDecoderLimitsDefaults(unsigned& mediaMaxWidth, unsigned& mediaMaxHeight, unsigned& mediaMaxFrameRate)
+{
+#ifdef VIDEO_DECODING_LIMIT
+    // VIDEO_DECODING_LIMIT should be in format: WIDTHxHEIGHT@FRAMERATE
+    String videoDecodingLimit(String::fromUTF8(VIDEO_DECODING_LIMIT));
+
+    if (videoDecodingLimit.isEmpty())
+        return;
+
+    Vector<String> entries;
+
+    // extract frame rate part from the VIDEO_DECODING_LIMIT: WIDTHxHEIGHT@FRAMERATE
+    videoDecodingLimit.split('@', [&entries](StringView item) {
+        entries.append(item.toString());
+    });
+
+    if (entries.size() != 2)
+        return;
+
+    mediaMaxFrameRate = parseIntegerAllowingTrailingJunk<unsigned>(entries[1]).value_or(0);
+
+    String widthAndHeight = entries[0];
+    entries.clear();
+
+    // extract WIDTH and HEIGHT from: WIDTHxHEIGHT
+    widthAndHeight.split('x', [&entries](StringView item) {
+        entries.append(item.toString());
+    });
+
+    if (entries.size() != 2)
+        return;
+
+    mediaMaxWidth = parseIntegerAllowingTrailingJunk<unsigned>(entries[0]).value_or(0);
+    mediaMaxHeight = parseIntegerAllowingTrailingJunk<unsigned>(entries[1]).value_or(0);
+#endif
```

Comment:
Everything has been successful until here, return true before the `#endif`. Return false in an `#else`, for those cases where `VIDEO_DECODING_LIMIT` isn't defined.

Reply:
Just to be sure your intention here: if we return false in #else (VIDEO_DECODING_LIMIT isn't defined) then we will get error message in that case too

Reply:
Hmmm... your're right. Maybe in that case it makes sense to return true and just leave the default initial values (zero). It doesn't make sense to bug the user in the logs about an error that isn't actually an error.